tests/spark_config_test.py

```diff
@@ -296,6 +296,55 @@ def gpu_pool(self, tmpdir, monkeypatch):
 
     @pytest.mark.parametrize(
         'cluster_manager,user_spark_opts,expected_output', [
+            # dynamic resource allocation enabled
+            (
+                'kubernetes',
+                {
+                    'spark.dynamicAllocation.enabled': 'true',
+                    'spark.executor.cores': '4',
+                    'spark.cores.max': '128',
```

Comment:
Let's have a test case without the config, `spark.cores.max`. This ideally should not be used in future.